docs/help/FAQ/Puppet.rst

```diff
@@ -3,57 +3,133 @@
+Situations :program:`incron` handles automatically
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, SIMP configures the :program:`incron` daemon to automatically run
+``puppet generate types`` under either of the following circumstances:
 
   * The ``puppet`` or ``puppetserver`` binaries have been updated.
-  * A new environment is added to the system.
+  * A new :term:`Puppet environment` directory is added to the system.
 
-You will need to run ``puppet generate types`` manually if the following occurs:
+This behavior is managed by the Puppet class ``pupmod::master::generate_types``.
 
-  * A new module is added to your environment that includes custom types.
-  * An existing custom type has its code modified.
+.. rubric:: Differences from Previous versions of SIMP
 
-.. NOTE::
-   Versions 7.6.0 through 7.7.1 of the ``simp-pupmod`` Puppet module tried
-   to include all of the cases above to ensure that users did not need to
-   manually adjust any aspects of their systems. However, this proved to
-   potentially add too much load to the system in certain situations and
-   was reduced to the current functionality.
+Earlier versions of :program:`simp-pupmod` (7.6.0 through 7.7.1, shipped with
+SIMP 6.2.0-0 through 6.3.1-0) attempted to automatically trigger ``puppet
+generate types`` under every relevant circumstance.  However, some of the
+triggers could add too much load on the system and were removed from the
+:program:`incron`'s watchlist.
+
+These situations must be addressed by other means (see below).
+
+
+Situations :program:`incron` doesn't handle
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:program:`incron` does not handle all cases, so you will need to ensure that
+``puppet generate types`` is after the following events:
+
+  * A new *module* that includes custom types is added to an existing environment.
+  * An existing custom type's internal code is updated.
+
+
+Generating types manually
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can run the ``puppet generate types`` command as **root** on the Puppet
+Server.  However, in order to ensure that the Puppet Server process can read
+the generated files, you must also ensure they have the correct ownership and
+permissions.  One way to do this is by running the following command:
+
+.. code-block:: bash
+
+   (umask 0027 && sg puppet -c 'puppet generate types --environment ENVIRONMENT')
+
+This creates all files with the correct group ownership.
+
+
+Automatically generating types after ``r10k deploy environment``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you are using :term:`r10k` to deploy :term:`Control Repository` branches
+using ``r10k deploy environment``, you can set the `"generate_types" option`_
+in the :file:`r10k.yaml` file to automatically run :command:`puppet generate
+types` for each environment after it is deployed:
+
+.. code-block:: yaml
+   :emphasize-lines: 3
+   :caption: Inside :file:`r10k.yaml`:
```

Comment:
caption does not seem to be a valid option.   It is producing warnings.

Reply:
@jeannegreulich I can't reproduce this warning on my local venv or in Travis CI―at least, not with the versions in `requirements.txt`.   What version of `sphinx-build` are you using?

----

I did get [this warning](https://travis-ci.org/simp/simp-doc/jobs/569028005#L589) (which I've fixed):
```
WARNING: Could not lex literal_block as "bash". Highlighting skipped.
```